CMakeLists.txt

```diff
@@ -47,25 +51,58 @@ if(${CUDA_VERSION} VERSION_GREATER_EQUAL "10.0")
 else()
   option(BUILD_NVTX "Build with NVTX profiling enabled" OFF)
 endif()
-option(BUILD_PYTHON "Build Python bindings" ON)
-option(BUILD_LMDB "Build LMDB readers" OFF)
-option(BUILD_JPEG_TURBO "Build with libjpeg-turbo support" ON)
-option(BUILD_LIBTIFF "Build with libtiff support" ON)
+
+# Third party library dependencies
+cmake_dependent_option(BUILD_PYTHON "Build Python bindings" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_LMDB "Build LMDB readers" OFF
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_JPEG_TURBO "Build with libjpeg-turbo support" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_LIBTIFF "Build with libtiff support" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_FFMPEG "Build with ffmpeg support" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_LIBSND "Build with suport for libsnd library" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_OPENCV "Build with suport for OpenCV library" ON
+                       "NOT NODEPS_BUILD" OFF)
+cmake_dependent_option(BUILD_PROTOBUF "Build with suport for Protobuf library" ON
+                       "NOT NODEPS_BUILD" OFF)
+option(BUILD_FFTS "Build with ffts support" ON)  # Built from thirdparty sources
+
+# NVIDIA libraries
+cmake_dependent_option(BUILD_NVDEC "Build with NVIDIA NVDEC support" ON
+                       "BUILD_FFMPEG" OFF)  # NVDEC support requires FFMPEG
 option(BUILD_NVJPEG "Build with nvJPEG support" ON)
 option(BUILD_NVOF "Build with NVIDIA OPTICAL FLOW SDK support" ON)
-option(BUILD_NVDEC "Build with NVIDIA NVDEC support" ON)
-option(BUILD_LIBSND "Build with suport for libsnd library" ON)
 option(BUILD_NVML "Build with NVIDIA Management Library (NVML) support" ON)
-option(BUILD_FFTS "Build with ffts support" ON)
-option(VERBOSE_LOGS "Adds verbose loging to DALI" OFF)
 
+# DALI modules
+# Note dali_core is always enabled
+option(BUILD_DALI_KERNELS "Build dali_kernels library" ON)
+cmake_dependent_option(BUILD_DALI_PIPELINE "Build dali pipeline library" ON
+                       "NOT NODEPS_BUILD;BUILD_DALI_KERNELS" OFF)
+cmake_dependent_option(BUILD_DALI_OPERATORS "Build dali operators library" ON
+                       "NOT NODEPS_BUILD;BUILD_DALI_KERNELS" OFF)
+
+# Experimental, only enabled for NODEPS_BUILD=ON
+cmake_dependent_option(STATIC_LIBS "Build static libraries instead of shared-object libraries" OFF
+                       "NODEPS_BUILD" OFF)
```

Comment:
Out of curiosity, couldn't we always build `dali_kernels` and `dali_core` as static libs?

Reply:
we could